### PR TITLE
Add buffer to progress

### PR DESCRIPTION
**Pull request by Fercas123**

### CODE REVIEW

**Review by ivan-calderon — DISMISSED** [review]

File: .changeset/tidy-carrots-allow.md [hunk 1]
```diff
@@ -0,0 +1,5 @@
+---
+"@salt-ds/core": minor
```
Comment (origami-z) [inline .changeset/tidy-carrots-allow.md:2]: Shall we talk about visual diff on circular?

File: packages/core/src/progress/LinearProgress/LinearProgress.tsx [hunk 2]
```diff
@@ -63,6 +78,7 @@ export const LinearProgress = forwardRef<HTMLDivElement, LinearProgressProps>(
       >
         <div className={withBaseName("barContainer")}>
           <div className={withBaseName("bar")} style={barStyle} />
+          <div className={withBaseName("buffer")} style={bufferStyle} />
```
Comment (origami-z) [inline packages/core/src/progress/LinearProgress/LinearProgress.tsx:81]: Could skip render when buffer is not there?

**Review by origami-z — COMMENTED** [review]
When looking at generated DOM, I don't quite get why the approach is complex/different for a relative simple component. For linear one, bar & track doing translate with total 200% width movement, buffer on the other hand doing width increment. Circular one could be more complex with left/right split.

**Comment by Fercas123** [comment]
> When looking at generated DOM, I don't quite get why the approach is complex/different for a relative simple component. For linear one, bar & track doing translate with total 200% width movement, buffer on the other hand doing width increment. Circular one could be more complex with left/right split.

I changed track to be static in linear, and value to grow in width rather than moving

File: packages/core/src/progress/LinearProgress/LinearProgress.css [hunk 3]
```diff
@@ -15,24 +15,37 @@
...
+  outline: solid var(--salt-size-border) var(--salt-accent-background);
+  z-index: var(--salt-zIndex-default);
+  outline-offset: calc(var(--salt-size-border) * -1);
+}
+
+.saltLinearProgress-bar {
+  width: 100%;
   background: var(--salt-accent-background);
-  z-index: 2;
+  z-index: calc(var(--salt-zIndex-default) + var(--salt-zIndex-default));
```
Comment (joshwooding) [inline packages/core/src/progress/LinearProgress/LinearProgress.css:39]:
```suggestion
  z-index: calc(var(--salt-zIndex-default) * 2);
```

**Comment by lukeac123** [comment]
NIT: Linear Progress has a min-width of 400px, not sure if there was a design decision behind this ? - but on smaller screen sizes this may have an impact. ( Doesn't really concern this pr so a change can follow up if needed )

**Review by lukeac123 — APPROVED** [review]

File: site/src/examples/progress/WithProgBufferVal.tsx [hunk 4]
```diff
@@ -0,0 +1,118 @@
...
+  LinearProgress,
+} from "@salt-ds/core";
+
+function useProgressingValue(updateInterval = 100) {
+  const [bufferValue, setBufferValue] = useState(0);
+  const [isProgressing, setIsProgressing] = useState(false);
+
+  const handleStop = useCallback(() => {
+    if (isProgressing) {
+      setIsProgressing(false);
```
Comment (joshwooding) [inline site/src/examples/progress/WithProgBufferVal.tsx:19]: Is this if statement even needed?

File: site/src/examples/progress/WithProgBufferVal.tsx [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+    }
+  }, [isProgressing]);
+
+  const handleStart = () => {
+    if (!isProgressing) {
```
Comment (joshwooding) [inline site/src/examples/progress/WithProgBufferVal.tsx:24]: same here

File: site/src/examples/progress/WithProgBufferVal.tsx [hunk 6]
```diff
@@ -0,0 +1,118 @@
...
+          <Button disabled={!isProgressing} onClick={handleStop}>
+            Stop
+          </Button>
+          <Button onClick={handleReset}>Reset</Button>
+        </FlowLayout>
+      </FlexItem>
+
+      <FlexItem>
+        <FlowLayout justify="center" className="controls" gap={1}>
+          <RadioButtonGroup direction="horizontal" defaultChecked>
```
Comment (joshwooding) [inline site/src/examples/progress/WithProgBufferVal.tsx:86]: This code isn't correct

Comment (Fercas123) [reply site/src/examples/progress/WithProgBufferVal.tsx:86]: in what way?

File: site/src/examples/progress/WithBuffer.tsx [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+export const WithBuffer = (): ReactElement => {
+  const [selectedType, setSelectedType] = useState("circular");
+
+  return (
+    <FlexLayout direction="column" style={{ height: "100%" }}>
+      <H3 style={{ textAlign: "center" }}> value = 38, buffer value = 60</H3>
+      <FlexItem>
+        <FlowLayout justify="center" className="controls" gap={1}>
+          <RadioButtonGroup direction="horizontal" defaultChecked>
```
Comment (joshwooding) [inline site/src/examples/progress/WithBuffer.tsx:21]: This code also needs to be looked at

File: site/src/examples/progress/WithBuffer.tsx [hunk 8]
```diff
@@ -0,0 +1,54 @@
...
+      <FlexItem align="center" grow={1}>
+        {selectedType === "circular" && (
+          <CircularProgress aria-label="Download" value={38} bufferValue={60} />
+        )}
+        {selectedType === "linear" && (
+          <LinearProgress
+            aria-label="Download"
+            value={38}
+            bufferValue={60}
+            style={{ height: "100%" }}
```
Comment (joshwooding) [inline site/src/examples/progress/WithBuffer.tsx:48]: Is this height needed?

File: site/src/examples/progress/WithProgBufferVal.tsx [hunk 9]
```diff
@@ -0,0 +1,118 @@
...
+
+      <FlexItem align="center" grow={1}>
+        {selectedType === "circular" && (
+          <CircularProgress aria-label="Download" bufferValue={bufferValue} />
+        )}
+        {selectedType === "linear" && (
+          <LinearProgress
+            aria-label="Download"
+            bufferValue={bufferValue}
+            style={{ height: "100%" }}
```
Comment (joshwooding) [inline site/src/examples/progress/WithProgBufferVal.tsx:112]: Is this height needed?

File: site/src/examples/progress/WithBuffer.tsx [hunk 10]
```diff
@@ -0,0 +1,54 @@
...
+} from "@salt-ds/core";
+
+export const WithBuffer = (): ReactElement => {
+  const [selectedType, setSelectedType] = useState("circular");
+
+  return (
+    <FlexLayout direction="column" style={{ height: "100%" }}>
+      <H3 style={{ textAlign: "center" }}> value = 38, buffer value = 60</H3>
+      <FlexItem>
+        <FlowLayout justify="center" className="controls" gap={1}>
```
Comment (joshwooding) [inline site/src/examples/progress/WithBuffer.tsx:20]: This className doesn't look like it does anything?

File: site/docs/components/progress/examples.mdx [hunk 11]
```diff
@@ -56,6 +56,20 @@ Specify the minimum value of the progress indicator. The percentage progress wil
...
+<LivePreview componentName="progress" exampleName="WithBuffer" displayName="With buffer">
+
+## With buffer
+
+Specify a buffer value to indicate loading state.
+
+</LivePreview>
+<LivePreview componentName="progress" exampleName="WithProgBufferVal" displayName="With progressing buffer value">
+
+## With progressing value
```
Comment (joshwooding) [inline site/docs/components/progress/examples.mdx:69]:
```suggestion
## With progressing buffer value
```

File: site/docs/components/progress/examples.mdx [hunk 12]
```diff
@@ -56,6 +56,20 @@ Specify the minimum value of the progress indicator. The percentage progress wil
 
 Dynamically represent a progressing value in the progress indicator.
 
+</LivePreview>
+<LivePreview componentName="progress" exampleName="WithBuffer" displayName="With buffer">
+
+## With buffer
+
+Specify a buffer value to indicate loading state.
```
Comment (joshwooding) [inline site/docs/components/progress/examples.mdx:64]: We should document that buffer doesn't affect the progress label.

**Review by lilyvc — DISMISSED** [review]

**Review by lukeac123 — APPROVED** [review]

**Pull request merged**